sdk/storage/Azure.Storage.Common/src/Shared/PooledMemoryStream.cs

```diff
@@ -54,13 +54,24 @@ private class BufferPartition
         /// </summary>
         private List<BufferPartition> BufferSet { get; } = new List<BufferPartition>();
 
+        /// <summary>
+        /// Size to rent from MemoryPool during Writes.
+        /// </summary>
+        public int ArrayPoolRentalSize { get; }
+
         private PooledMemoryStream(ArrayPool<byte> arrayPool, long absolutePosition, int maxArraySize)
         {
             AbsolutePosition = absolutePosition;
             ArrayPool = arrayPool;
             MaxArraySize = maxArraySize;
         }
 
+        public PooledMemoryStream(int arrayPoolRentalSize)
```

Comment:
If we consolidate `MaxArraySize` and `ArrayPoolRentalSize` then maybe we should consolidate these constructors. `arrayPool` and `absolutePosition` could be default parameters.

Reply:
Will do.